### ISSUE TITLE
Build of latest version of smali 2.2_WIP proguard failed

### ISSUE THREAD

**Issue by bursoft-git** [JesusFreke/smali#405]
```
C:\smali-db49ae1d03e918a21c2ce7afe49a5ce841d950a8>java -version
java version "1.7.0_101"
Java(TM) SE Runtime Environment (build 1.7.0_101-b14)
Java HotSpot(TM) 64-Bit Server VM (build 24.101-b14, mixed mode)

C:\smali-db49ae1d03e918a21c2ce7afe49a5ce841d950a8>gradlew proguard
Parallel execution is an incubating feature.
:util:compileJava
:baksmali:processResources
:dexlib2:processResources UP-TO-DATE
:smali:generateGrammarSource
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
:util:processResources UP-TO-DATE
:util:classes
:util:jar
:dexlib2:compileJava
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
:dexlib2:classes
:dexlib2:jar
:baksmali:compileJava
:baksmali:classes
:baksmali:fatJar
:smali:generateJFlexSource
:smali:compileJava
:baksmali:proguard
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
:smali:processResources
:smali:classes
:smali:jar
:smali:fatJar
:smali:proguard
:baksmali:proguard FAILED
Reading library jar [C:\JDK\jre\lib\rt.jar]
Note: the configuration refers to the unknown class 'org.jf.smali.main'
Note: com.beust.jcommander.JCommander calls 'Class.getAnnotation'
Note: com.beust.jcommander.ParameterDescription calls 'Class.getAnnotation'
Note: com.beust.jcommander.Parameterized calls 'Field.getAnnotation'
Note: com.beust.jcommander.Parameterized calls 'Method.getAnnotation'
Note: org.jf.util.jcommander.Command calls 'Class.getAnnotation'
Note: org.jf.util.jcommander.ExtendedCommands calls 'Class.getAnnotation'
Note: org.jf.util.jcommander.ExtendedCommands calls 'Field.getAnnotation'
Note: org.jf.util.jcommander.HelpFormatter calls 'Class.getAnnotation'
Note: com.beust.jcommander.Parameterized calls 'Field.getType'
Note: com.beust.jcommander.Parameterized calls 'Method.getGenericParameterTypes'
Note: com.beust.jcommander.Parameterized calls 'Field.getGenericType'
Note: org.jf.dexlib2.analysis.reflection.ReflectionField calls 'Field.getType'
Note: com.beust.jcommander.WrappedParameter accesses a method 'put(java.lang.Object,java.lang.Object)' dynamically
      Maybe this is library method 'com.sun.beans.WeakCache { void put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'com.sun.corba.se.impl.util.IdentityHashtable { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'com.sun.java.util.jar.pack.PropMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'com.sun.jmx.mbeanserver.WeakIdentityHashMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'com.sun.net.httpserver.Headers { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'com.sun.org.apache.xalan.internal.xsltc.compiler.util.MultiHashtable { java.util.Set put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'com.sun.org.apache.xerces.internal.util.SymbolHash { void put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'com.sun.script.util.BindingsBase { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'com.sun.xml.internal.bind.v2.schemagen.MultiMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'com.sun.xml.internal.ws.client.RequestContext$MapView { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'com.sun.xml.internal.ws.client.ResponseContext { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'com.sun.xml.internal.ws.handler.MessageContextImpl { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'com.sun.xml.internal.ws.handler.MessageUpdatableContext { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'com.sun.xml.internal.ws.server.EndpointMessageContextImpl { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.awt.RenderingHints { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.lang.ProcessEnvironment { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.security.Provider { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.AbstractMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.Collections$CheckedMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.Collections$SynchronizedMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.Collections$UnmodifiableMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.Dictionary { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.EnumMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.HashMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.Hashtable { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.IdentityHashMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.Map { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.TreeMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.TreeMap$NavigableSubMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.WeakHashMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.concurrent.ConcurrentHashMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.concurrent.ConcurrentSkipListMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.concurrent.ConcurrentSkipListMap$SubMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'java.util.jar.Attributes { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'javax.management.openmbean.TabularDataSupport { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'javax.print.attribute.standard.PrinterStateReasons { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'javax.script.SimpleBindings { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'javax.swing.ArrayTable { void put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'javax.swing.UIDefaults { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'javax.swing.UIManager { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'javax.xml.crypto.XMLCryptoContext { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'javax.xml.crypto.dom.DOMCryptoContext { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'jdk.xml.internal.AbstractMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'jdk.xml.internal.HashMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'sun.awt.AppContext { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'sun.awt.AppContext$6 { void put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'sun.awt.FontConfiguration$PropertiesHandler$FontProperties { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'sun.font.AttributeMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'sun.misc.Cache { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'sun.misc.JavaAWTAccess { void put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'sun.misc.SoftCache { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'sun.net.httpserver.UnmodifiableHeaders { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'sun.org.mozilla.javascript.internal.NativeObject { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'sun.security.util.Cache { void put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'sun.security.util.MemoryCache { void put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'sun.security.util.NullCache { void put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'sun.util.PreHashedMap { java.lang.Object put(java.lang.Object,java.lang.Object); }'
      Maybe this is library method 'sun.util.locale.LocaleObjectCache { java.lang.Object put(java.lang.Object,java.lang.Object); }'
Note: com.beust.jcommander.internal.JDK6Console accesses a declared method 'writer()' dynamically
      Maybe this is library method 'java.io.Console { java.io.PrintWriter writer(); }'
Note: com.beust.jcommander.internal.JDK6Console accesses a declared method 'readLine()' dynamically
      Maybe this is library method 'com.sun.corba.se.impl.io.IIOPInputStream { java.lang.String readLine(); }'
      Maybe this is library method 'com.sun.xml.internal.messaging.saaj.packaging.mime.util.LineInputStream { java.lang.String readLine(); }'
      Maybe this is library method 'com.sun.xml.internal.org.jvnet.mimepull.MIMEParser$LineInputStream { java.lang.String readLine(); }'
      Maybe this is library method 'java.io.BufferedReader { java.lang.String readLine(); }'
      Maybe this is library method 'java.io.Console { java.lang.String readLine(); }'
      Maybe this is library method 'java.io.DataInput { java.lang.String readLine(); }'
      Maybe this is library method 'java.io.DataInputStream { java.lang.String readLine(); }'
      Maybe this is library method 'java.io.LineNumberReader { java.lang.String readLine(); }'
      Maybe this is library method 'java.io.ObjectInputStream { java.lang.String readLine(); }'
      Maybe this is library method 'java.io.ObjectInputStream$BlockDataInputStream { java.lang.String readLine(); }'
      Maybe this is library method 'java.io.RandomAccessFile { java.lang.String readLine(); }'
      Maybe this is library method 'javax.imageio.stream.ImageInputStream { java.lang.String readLine(); }'
      Maybe this is library method 'javax.imageio.stream.ImageInputStreamImpl { java.lang.String readLine(); }'
      Maybe this is library method 'sun.net.httpserver.Request { java.lang.String readLine(); }'
Note: com.beust.jcommander.internal.JDK6Console accesses a declared method 'readPassword()' dynamically
      Maybe this is library method 'java.io.Console { char[] readPassword(); }'
Note: there were 1 references to unknown classes.
      You should check your configuration for typos.
      (http://proguard.sourceforge.net/manual/troubleshooting.html#unknownclass)
Note: there were 8 classes trying to access annotations using reflection.
      You should consider keeping the annotation attributes
      (using '-keepattributes *Annotation*').
      (http://proguard.sourceforge.net/manual/troubleshooting.html#attributes)
Note: there were 4 classes trying to access generic signatures using reflection.
      You should consider keeping the signature attributes
      (using '-keepattributes Signature').
      (http://proguard.sourceforge.net/manual/troubleshooting.html#attributes)
Note: there were 4 accesses to class members by means of introspection.
      You should consider explicitly keeping the mentioned class members
      (using '-keep' or '-keepclassmembers').
      (http://proguard.sourceforge.net/manual/troubleshooting.html#dynamicalclassmember)
:smali:proguard FAILED

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':baksmali:proguard'.
> java.io.IOException: The output jar is empty. Did you specify the proper '-keep' options?

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':smali:proguard'.
> java.io.IOException: The output jar is empty. Did you specify the proper '-keep' options?

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
==============================================================================

BUILD FAILED

Total time: 5.469 secs
```

**Comment by Aex12** [comment]
Run `gradlew build --stacktrace` to get the stack trace, so the devs can get more info about your error

**Comment by bursoft-git** [comment]
Here is log with --stacktrace
[gradlew proguard --stacktrace.txt](https://github.com/JesusFreke/smali/files/313260/gradlew.proguard.--stacktrace.txt)

**Comment by JesusFreke** [comment]
Fixed: c2231759
